### DISCUSSION TITLE
有无朋友在f103c8t6上面复刻了，能不能交流一下呢

### DISCUSSION THREAD

**Discussion by Chasen-Liao** [AstraThreshold/oled-ui-astra#10] [Q&A]
我的HAL库不是很明白要怎么去改写，改写完之后，astra能识别吗，还是要另外写？

**Comment by LIU-ZI-Giter** [comment]
我使用标准库开发f103c8t6，目前底层移植完成，可以正常播放开机动画，但是用部署教程中的程序进行测试发现，Menu类无法正常使用，会导致程序进入HardFault，从而卡死。

**Reply by EnumerationCat** [reply]
指针的问题，出现空指针，删除导致的指针悬挂

**Reply by EnumerationCat** [reply]
把内存管理的.c.h删掉

**Comment by leisure9527** [comment]
我在F722上移植成功了，并且借助本身速度1000k的i2c plus，虽然是i2c但刷屏极快

**Comment by luo96997** [comment]
c8t6在keil上移植成功了，还不知道怎么用，现在正在摸索

**Reply by 5akura1** [reply]
朋友能不能交流一下，我也是keil，可以正常播放开机动画，但是用部署教程中的程序发现Menu类无法正常使用，会导致程序进入HardFault，然后从而卡死。还望指导！！

**Reply by LIU-ZI-Giter** [reply]
后面我换用了另一个更轻量的ui，时间有点久了，我记得导致hardfault的原因好像是栈溢出，通过修改启动文件的堆栈空间可以暂时解决。

**Reply by luo96997** [reply]
> 朋友能不能交流一下，我也是keil，可以正常播放开机动画，但是用部署教程中的程序发现Menu类无法正常使用，会导致程序进入HardFault，然后从而卡死。还望指导！！

这个应该是C8T6本身的Flash内存不够了，因为运行基础的程序之后就已经90%多占用了，再加入图片的话就不够运行的了，这样导致崩溃报硬件错误

**Reply by luo96997** [reply]
> 后面我换用了另一个更轻量的ui，时间有点久了，我记得导致hardfault的原因好像是栈溢出，通过修改启动文件的堆栈空间可以暂时解决。
C8T6移植这个不太合适，有能力的话可以上一个Flash大一点的芯片

**Comment by leisure9527** [comment]
这个作者都已经出更轻量化的ui了发自我的荣耀手机-------- 原始邮件 --------发件人： luo96997 ***@***.***>日期： 2025年6月25日周三 15:38收件人： AstraThreshold/oled-ui-astra ***@***.***>抄送： 张月月 ***@***.***>, Comment ***@***.***>主    题： Re: [AstraThreshold/oled-ui-astra] 有无朋友在f103c8t6上面复刻了，能不能交流一下呢 (Discussion #10)

朋友能不能交流一下，我也是keil，可以正常播放开机动画，但是用部署教程中的程序发现Menu类无法正常使用，会导致程序进入HardFault，然后从而卡死。还望指导！！

这个应该是C8T6本身的Flash内存不够了，因为运行基础的程序之后就已经90%多占用了，再加入图片的话就不够运行的了，这样导致崩溃报硬件错误

**Reply by luo96997** [reply]
是的，但是现在这个也能在C8T6运行，还好

**Comment by leisure9527** [comment]
我当时在f722上面部署过，特别流畅发自我的荣耀手机-------- 原始邮件 --------发件人： luo96997 ***@***.***>日期： 2025年6月25日周三 16:08收件人： AstraThreshold/oled-ui-astra ***@***.***>抄送： 张月月 ***@***.***>, Comment ***@***.***>主    题： Re: [AstraThreshold/oled-ui-astra] 有无朋友在f103c8t6上面复刻了，能不能交流一下呢 (Discussion #10)
是的，但是现在这个也能在C8T6运行，还好
